dask/dataframe/utils.py

```diff
@@ -734,3 +734,49 @@ def drop_by_shallow_copy(df, columns, errors="raise"):
 
 class AttributeNotImplementedError(NotImplementedError, AttributeError):
     """NotImplementedError and AttributeError"""
+
+
+def meta_frame_constructor(like):
```

Comment:
Can we add some basic test coverage for `meta_frame_constructor` / `meta_series_constructor` to `dask/dataframe/tests/test_utils_dataframe.py`? In particular it'd be good to test that, for `meta_frame_constructor`,  `pd.DataFrame` / `pd.Series` / `pd.Index` return `pd.DataFrame` and we raise a `TypeError` when we expect (e.g. when a `np.ndarray` in used as input). Same for their gpu counterparts.

Reply:
Sounds good!  Just added.